docs/docs/architecture/security.md

```diff
@@ -0,0 +1,97 @@
+Furthermore, MarbleRun's' [RBAC](../workflows/define-manifest.md#roles) attaches a Marble's identity to a role in the manifest.
+Each role is associated with a set of operations that Marble can perform in the deployment.
+Roles can also be attached to [users](../workflows/define-manifest.md#users), which are authenticated by the Coordinator using an RSA or ECDSA public key defined in the manifest.
+
+## Public Key Infrastructure and Certificate Authority
+
+The Coordinator establishes a public key infrastructure (PKI) for MarbleRun and acts as the Certificate Authority (CA).
+The goal of the PKI is to make authentication of confidential applications based on remote attestation accessible and usable.
+The Coordinator embeds its attestation statement into its root CA certificate, see the [attested TLS](#attested-tls-atls) section for details behind that concept.
+All MarbleRun clients and Marbles can then use the attested root CA certificate for authenticating TLS connections.
+This is further illustrated conceptionally in the [attestation](../features/attestation.md) section. We now focus on the cryptography.
+
+During initialization the Coordinator generates a root x509 certificate and corrisponding asymmetric key pair.
+The [Elliptic Curve Digital Signature Algorithm (ECDSA)](https://www.secg.org/sec1-v2.pdf#page=49) is used with curve [P256](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf#page=111).
+The root certificate has no expiary date set and lives as long as the MarbleRun deployment.
+
+Alongside the root certificate, the Coordinator generates an intermediate x509 certificate and corresponding asymmetric key pair, again using ECDSA with P256.
+The intermediate certificate is signed by the Coordinator's root certificate and rotated with every update of the manifest.
+When you push an update to the manifest (for example bump up the *SecurityVersion* of a Mable) the intermediate certificate will change.
+Instances with the new version will not authenticate with instances of the old version and vice versa.
+Hence, no data flow is happening between different *SecurityVersions* of your application.
+However, the root certificate doesn't change. So you can still verify the Coordinator and your application from the outside and make sure it is the same instance you might have interacted with before.
+
+The Coordinator creates a second intermediate certificate that is called the *Marble root certificate*.
+The Marble root certificate is self-signed using the intermediate certificate's private key implementing a  [cross-signed certificate chain](https://www.ssltrust.com.au/blog/understanding-certificate-cross-signing).
+In that way, the Marbles see a self-signed root certificate, hence, the are dealing with a terminating certificate chain without knowing about the Coordinator's root certificate.
+The "outside world" sees an intermediate certificate signed by the Coordinator's root certificate and the corresponding root certificate.
```

Comment:
This may be hard to comprehend. Not sure how to improve it without making it much more verbose, though.

Reply:
Tried to explain it a bit more verbose, not sure if that helps?